config/opal_check_package2.m4

```diff
@@ -0,0 +1,232 @@
+dnl -*- autoconf -*-
+dnl
+dnl Copyright (c) 2004-2005 The Trustees of Indiana University and Indiana
+dnl                         University Research and Technology
+dnl                         Corporation.  All rights reserved.
+dnl Copyright (c) 2004-2005 The University of Tennessee and The University
+dnl                         of Tennessee Research Foundation.  All rights
+dnl                         reserved.
+dnl Copyright (c) 2004-2005 High Performance Computing Center Stuttgart,
+dnl                         University of Stuttgart.  All rights reserved.
+dnl Copyright (c) 2004-2005 The Regents of the University of California.
+dnl                         All rights reserved.
+dnl Copyright (c) 2012-2017 Cisco Systems, Inc.  All rights reserved.
+dnl Copyright (c) 2012      Oracle and/or its affiliates.  All rights reserved.
+dnl Copyright (c) 2014      Intel, Inc. All rights reserved.
+dnl Copyright (c) 2015-2018 Research Organization for Information Science
+dnl                         and Technology (RIST). All rights reserved.
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+dnl OPAL_DECLARE_PACKAGE(name,[help],[cppflagshelp],[ldflagshelp])
+AC_DEFUN([OPAL_DECLARE_PACKAGE],[
+    AC_ARG_WITH([$1], [AC_HELP_STRING([--with-$1=DIR], [$2])])
+    AC_ARG_WITH([$1-cppflags], [AC_HELP_STRING([--with-$1-cppflags=FLAGS], [$3])])
+    AC_ARG_WITH([$1-ldflags], [AC_HELP_STRING([--with-$1-ldflags=FLAGS], [$4])])
+    AC_ARG_WITH([$1-libdir], [AC_HELP_STRING([--with-$1-libdir=DIR],
+                                             [Deprecated, use --with-$1-ldflags instead])])
+    AS_IF([test -n "$with_$1_libdir"],
+          [AS_IF([test -n "$with_$1_ldflags"],
+                 [AC_MSG_WARN([It is not possible to use both --with-$1-ldflags and --with-$1-libdir])
+                  AC_MSG_ERROR([Cannot continue])],
+                 [AC_MSG_WARN([--with_$1_libdir=DIR is deprecated, use --with-$1_ldflags=FLAGS instead])
+                  with_$1_ldflags="-L$with_$1_libdir"])])
+])
+
+dnl --------------------------------------------------------------------
+dnl _OPAL_CHECK_PACKAGE2_HEADER(package, prefix, header, includes,
+dnl                             [action-if-found], [action-if-not-found])
+dnl --------------------------------------------------------------------
+AC_DEFUN([_OPAL_CHECK_PACKAGE2_HEADER], [
+    AS_VAR_PUSHDEF([opal_With], [with_$1])
```

Comment:
This capitalization kind of makes me ragey.  Why the semi-CamelCase, with underscores?

Reply:
`opal_Lib` and `opal_Header` have been around for a while, but I get your point and will update this too.